juju/juju
# Add logging context for workers.

In the past workers that ran on the controller had all the logging go into the log collection of the controller model. That information was also written into the machine-x.log file.

This PR adds a new logging context for model workers started by the controller on behalf of a model.

This logging context has two log writers:
 * a file based writer that goes to /var/log/juju/models/<uuid>.log
 * a db log writer that writes into the model's log collection

## QA steps

 * Bootstrap a controller
 * deploy something into the default model, I used ubuntu-lite
 * add a few models and deploy things into those models too

SSH into the controller machine and check the /var/log/juju directory. Look at the logs in the appropriate model's log files.

Use the debug-log command on the client to see that the log entries ended up in the right model's collection.

## Documentation changes

No real documentation changes needed.

## Bug reference

I'm sure there is one somewhere.
